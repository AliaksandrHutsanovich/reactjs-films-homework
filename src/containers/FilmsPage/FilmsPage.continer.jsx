import React, {
  memo,
  useState,
  useCallback,
} from 'react';
import { useSelectionOperation } from '../../hooks';
import { SEARCH_VALUE_SELECTOR_TITLE } from '../../constants';
import FilmsPage from '../../pages';

const FilmsPageContainer = () => {
  const [withBackground, setWithBackground] = useState(true);
  const [scrollerRef, setScrollerRef] = useState(null);
  const handleSearch = useSelectionOperation(SEARCH_VALUE_SELECTOR_TITLE);
  const handleScroll = useCallback(
    ({ target }) => {
      if (!withBackground && !target.scrollTop) {
        setWithBackground(true);
      } else if (withBackground && target.scrollTop) {
        setWithBackground(false);
      }
    },
    [withBackground], //eslint-disable-line
  );
  const setRef = useCallback(setScrollerRef, [setScrollerRef]);

  return (
    <FilmsPage
      withBackground={withBackground}
      scrollerRef={scrollerRef}
      onSearch={handleSearch}
      onScroll={handleScroll}
      setRef={setRef}
    />
  );
};

export default memo(FilmsPageContainer);
