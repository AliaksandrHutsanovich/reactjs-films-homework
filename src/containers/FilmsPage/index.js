export { default } from './FilmsPage.continer';
