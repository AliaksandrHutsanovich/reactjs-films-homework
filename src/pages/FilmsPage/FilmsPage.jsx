import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { Layout } from 'antd';
import PageHeader from './PageHeader';
import MainInfoSection from './MainInfoSection';
import MoviesSection from './MoviesSection';
import PageFooter from './PageFooter';
import { useSelectionOperation } from '../../hooks';
import { SEARCH_VALUE_SELECTOR_TITLE } from '../../constants';

import styles from './FilmsPage.scss';

const FilmsPage = ({ onAction }) => {
  const handleSearch = useSelectionOperation(SEARCH_VALUE_SELECTOR_TITLE);

  return (
    <Layout className={styles.wrapper}>
      <PageHeader onAction={handleSearch} />
      <MainInfoSection onAction={onAction} />
      <MoviesSection />
      <PageFooter />
    </Layout>
  );
};

FilmsPage.propTypes = {
  onAction: PropTypes.func.isRequired,
};

export default memo(FilmsPage);
