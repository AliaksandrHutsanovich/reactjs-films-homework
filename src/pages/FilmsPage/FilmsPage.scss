%background-shared {
  background: url('../../assets/images/background.png') no-repeat;
  // background-size: calc(100% + 180px);
  background-size: 1330px 560px;
  background-position-x: -82px;
}

.wrapper {
  width: 1180px;
  margin: 0 auto;
}

.header {
  @extend %background-shared;
  background-position-y: -4px;
  display: flex;
  height: 50px;
  justify-content: space-between;
  padding: 10px 80px 10px 40px;
}

.layout {
  @extend %background-shared;
  background-position-y: -54px;
  height: 462px;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  padding-left: 40px;
  padding-right: 20px;
}

.listWrapper {
  width: auto;
}

.infoWrapper {
  display: flex;
  align-items: center;
  padding-bottom: 10px;
}

.rightCol {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  padding-bottom: 40px;
}

.leftCol {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
}

.textWrapper {
  width: 230px;
  height: auto;
  background-color: transparent;
  margin-bottom: 20px;
}

.buttonsWrapper {
  padding-bottom: 30px;
  padding-right: 20px;

  > button {
    margin: 0 5px;
  }
}

.footer {
  background-color: white;
  padding: 50px 50px;
  display: flex;
  flex-direction: column;
  align-items: center;
}
