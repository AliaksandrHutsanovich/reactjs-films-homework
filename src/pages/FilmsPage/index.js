export { default } from './FilmsPage';  //eslint-disable-line
