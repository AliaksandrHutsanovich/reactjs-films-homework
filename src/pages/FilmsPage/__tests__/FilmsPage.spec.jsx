import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import FilmsPage from '../FilmsPage';

jest.mock('../../../hooks/useSearch', () => () => () => {});

describe('<FilmsPage />', () => {
  it('should be rendered properly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <FilmsPage onAction={() => {}} />,
    );
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
