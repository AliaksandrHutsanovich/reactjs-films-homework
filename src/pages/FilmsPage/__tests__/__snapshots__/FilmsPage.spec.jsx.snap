// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmsPage /> should be rendered properly 1`] = `
<Layout
  className="wrapper"
>
  <UNDEFINED
    onAction={[Function]}
  />
  <UNDEFINED
    onAction={[Function]}
  />
  <UNDEFINED />
  <UNDEFINED />
</Layout>
`;
